lib/puppet/provider/package/yum.rb

```diff
@@ -106,7 +106,13 @@ def self.parse_updates(str)
   end
 
   def self.update_to_hash(pkgname, pkgversion)
-    name, arch = pkgname.split('.')
+    
+    # The pkgname string has two parts: name, and architecture. Architecture
+    # is the portion of the string following the last "." character. All
+    # characters preceding the final dot are the package name. Parse out
+    # these two pieces of component data.
+    name, _, arch = pkgname.rpartition('.')
+    raise "Failed to parse package name and architecture from '#{pkgname}'" if name.empty?    
```

Comment:
Can you make this message translatable?

```ruby
raise _("Failed to ... '%{pkgname}'") % { pkgname: pkgname }
```

See https://github.com/puppetlabs/puppet/blob/master/CONTRIBUTING.md#writing-translatable-code